syntax = "proto3";

package Cliente_Productor;

message Mensaje{
    int32 tema = 1;
    int32 idMensaje = 2; 
    string contenido = 3;

}

message estadoMSJ{
    bool recibido = 1;
 }

 message Vacio{}

 message solicitudMSJ{
    int32 tema = 1;
    int32 idMensaje = 2;
 }


service ClienteServicios{
    rpc enviarMensaje(Mensaje) returns (estadoMSJ) {}
    rpc publicarMensaje(Mensaje) returns(estadoMSJ) {}
    rpc recibirMSJServer(solicitudMSJ) returns (Mensaje) {}
}